notioncli/cli.py

```diff
@@ -0,0 +1,170 @@
+    cprint('  # Status Description','white',attrs=['underline'])
+    for child in page.children:
+        if child.type == 'sub_header':
+            cprint('[{}]'.format(child.title), 'green')
+        elif child.type == 'to_do':
+            n += 1
+            if child.checked:
+                check = '[*]'
+            else:
+                check = '[ ]'
+            cprint('  {}  {}  {}.'.format(n, check, child.title), 'green')
+        else:
+            pass    
+
+    cprint('\n{} total tasks'.format(n), 'white', attrs=['bold'])
+
+def check_task(page, taskn):
+    n = 0
+    for child in page.children:
+        if child.type == 'to_do':
+            n += 1
+            for task in taskn:
+                if n == int(task):
+                    child.checked = True
+        else:
+            pass #not a task 
+    cprint('{} marked as completed'.format(taskn), 'white', attrs=['bold'])
+
+def uncheck_task(page, taskn):
+    if isinstance(taskn, int):
+        taskn = str(taskn)
+    else:
+        if ',' in taskn:
+            taskn = taskn.split(',')
+    n = 0
+    for child in page.children:
+        n += 1
+        try:
+            for task in taskn:
+                if n == int(task):
+                    child.checked = False
+        except:
+            pass #not a task 
+    cprint('{} marked as incomplete'.format(taskn), 'white', attrs=['bold'])
+
+def add_task(page, task):
+    newchild = page.children.add_new(TodoBlock, title=task)
+    newchild.checked = False
+    cprint('{} added as a new task'.format(task))
+
+def remove_task(page, taskn):
+
+    if isinstance(taskn, int):
+        taskn = str(taskn)
+    else:
+        if ',' in taskn:
+            taskn = taskn.split(',')
+    n = 0
+    for child in page.children:
+        if child.type =='to_do':
+            n += 1
+            for task in taskn:
+                if n == int(task):
+                    child.remove()      
+        else:
+            pass #not a task 
+    cprint('{} removed.'.format(taskn), 'white', attrs=['bold'])
+
+
+def main():
+    parser = argparse.ArgumentParser(description='A Notion.so CLI \
+    focused on simple task management')
+    parser.add_argument('--env', nargs='?', const=True, default=False, help='Print current relevant environment variables')
+    parser.add_argument('--list', nargs='?', const=True, default=False, help='List tasks')
+    parser.add_argument('--add', default=False, type=str, help='Usage: --add [str] Add a new task')
+    parser.add_argument('--remove', default=False, type=parse_task, help='Usage: --remove [n] (or n,n) remove task n from the task list')
+    parser.add_argument('--check', default=False, type=parse_task, help='Usage: --check [n] (or n,n) mark n as completed')
+    parser.add_argument('--uncheck', default=False, type=parse_task, help='Usage: --uncheck[n] (or n,n) to mark as incomplete')
+
+    args = parser.parse_args()
+
+    try:
+
+        check_env()
+
+        page = get_page(connect())
```

Comment:
Since all the commands have to make the client and load the page, I centralized all that logic here and pass it to the individual tasks.

Reply:
👍